apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js

```diff
@@ -0,0 +1,299 @@
+const { COUNTRIES, DEFAULT_LOCALE } = require('@condo/domains/common/constants/countries')
+
+const { Meter, MeterReading, MeterResource } = require('@condo/domains/meter/utils/serverSchema')
+const { Organization } = require('@condo/domains/organization/utils/serverSchema')
+
+const { sendMessage } = require('@condo/domains/notification/utils/serverSchema')
+const {
+    METER_SUBMIT_READINGS_REMINDER_TYPE,
+    METER_VERIFICATION_DATE_EXPIRED_TYPE,
+} = require('@condo/domains/notification/constants/constants')
+
+const { rightJoin, joinResidentsToMeters } = require('@condo/domains/meter/tasks/sendVerificationDateReminder')
+
+const logger = pino({
+    name: 'submit_meter_readings_notifications',
+    enabled: falsey(process.env.DISABLE_LOGGING),
+})
+
+const log = (message, intentChar = null, intentRepeatTimes = 10) => {
+    const logEntry = isNull(intentChar) ? message
+        : `${ intentChar.repeat(intentRepeatTimes) } ${ message } ${ intentChar.repeat(intentRepeatTimes) }`
+
+    logger.info({
+        message: logEntry,
+    })
+}
+
+const readMetersPage = async ({ context, offset, pageSize }) => {
+    return await Meter.getAll(
+        context, {}, {
+            sortBy: 'id_ASC',
+            first: pageSize,
+            skip: offset,
+        }
+    )
+}
+
+const readMeterReadings = async ({ context, meters }) => {
+    // first step is get unique meter ids
+    const meterIds = uniq(meters.map(meter => meter.id))
+
+    // then calculate current period window
+    const now = dayjs()
+    const startWindowDate = `${ now.format('YYYY-MM') }-01`
+    const endWindowDate = now.toISOString()
+
+    // load all pages for entity
+    return await loadListByChunks({
+        context,
+        list: MeterReading,
+        where: {
+            date_gte: startWindowDate,
+            date_lte: endWindowDate,
+            meter: {
+                id_in: meterIds,
+            },
+        },
+    })
+}
+
+const readOrganizations = async ({ context, meters }) => {
+    // first step is get unique organizations ids
+    const organizationIds = uniq(meters.map(meter => meter.organization.id))
+
+    // load all pages for entity
+    return await loadListByChunks({
+        context,
+        list: Organization,
+        where: {
+            id_in: organizationIds,
+        },
+    })
+}
+
+const getLocalizedMeterResources = async ({ context, lang }) => {
+    // get context for specific locale
+    const localeDependedContext = {
+        ...context.createContext({ skipAccessControl: true }),
+        req: { locale: lang },
+    }
+
+    // get list of meter resources for certain locale
+    // TODO apply DOMA-3174 mechanic for custom request locale
+    return await loadListByChunks({
+        context,
+        list: MeterResource,
+        where: {},
+    })
+}
+
+const loadAllLocalizedVersionsOfMeterResources = async ({ context }) => {
+    const supportedLanguages = ['en', 'ru']
```

Comment:
your code should not be painful to add new language. You assumption that we will change your function if need to add another language is wrong

Reply:
will fix